usage/src/org/apache/cloudstack/quota/QuotaManagerImpl.java

```diff
@@ -0,0 +1,465 @@
+                    usageRecord.getStartDate(), usageRecord.getEndDate());
+            _quotaUsageDao.persist(quota_usage);
+        }
+
+        usageRecord.setQuotaCalculated(1);
+        _usageDao.persist(usageRecord);
+        return quota_usage;
+    }
+
+    @DB
+    private QuotaUsageVO updateQuotaRaw(UsageVO usageRecord, final BigDecimal aggregationRatio, final int ruleType) {
+        QuotaUsageVO quota_usage = null;
+        QuotaTariffVO tariff = _quotaTariffDao.findTariffPlanByUsageType(ruleType, usageRecord.getEndDate());
+        if (tariff != null && !tariff.getCurrencyValue().equals(0)) {
+            BigDecimal ruleusage;
+            BigDecimal onehourcost;
+            onehourcost = tariff.getCurrencyValue().multiply(aggregationRatio);
+            // s_logger.info("Quotatariff onehourcost=" + onehourcost);
+            ruleusage = new BigDecimal(usageRecord.getRawUsage()).multiply(onehourcost);
+            quota_usage = new QuotaUsageVO(usageRecord.getId(), usageRecord.getZoneId(), usageRecord.getAccountId(), usageRecord.getDomainId(), ruleType, ruleusage, usageRecord.getStartDate(),
+                    usageRecord.getEndDate());
+            _quotaUsageDao.persist(quota_usage);
+        }
+
+        usageRecord.setQuotaCalculated(1);
+        _usageDao.persist(usageRecord);
+        return quota_usage;
+    }
+
+    @DB
+    private QuotaUsageVO updateQuotaNetwork(UsageVO usageRecord, final int transferType) {
+        QuotaUsageVO quota_usage = null;
+        QuotaTariffVO tariff = _quotaTariffDao.findTariffPlanByUsageType(transferType, usageRecord.getEndDate());
+        if (tariff != null && !tariff.getCurrencyValue().equals(0)) {
+            BigDecimal onegbcost;
+            BigDecimal rawusageingb;
+            BigDecimal networkusage;
+            onegbcost = tariff.getCurrencyValue();
+            // s_logger.info("Quotatariff onegbcost=" + onegbcost);
+            rawusageingb = new BigDecimal(usageRecord.getRawUsage()).divide(s_gb, 8, RoundingMode.HALF_EVEN);
+            networkusage = rawusageingb.multiply(onegbcost);
+            quota_usage = new QuotaUsageVO(usageRecord.getId(), usageRecord.getZoneId(), usageRecord.getAccountId(), usageRecord.getDomainId(), transferType, networkusage, usageRecord.getStartDate(),
+                    usageRecord.getEndDate());
+            _quotaUsageDao.persist(quota_usage);
+        }
+
+        usageRecord.setQuotaCalculated(1);
+        _usageDao.persist(usageRecord);
+        return quota_usage;
+    }
+
+    public Date startOfNextDay() {
+        Calendar c = Calendar.getInstance();
+        c.setTime(new Date());
+        c.add(Calendar.DATE, 1);
+        Date dt = c.getTime();
+        return dt;
+    }
+
+    protected boolean lockAccount(long accountId) {
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        TransactionLegacy.open(TransactionLegacy.CLOUD_DB).close();
```

Comment:
LegacyTranscaction is in use in cloud almost everywhere. While this is not a reason to continue its use. My thinking is that this should be delinked from Quota changes and looked at in the right perspective.
We should make a joint effort to clear this up.